core/src/main/scala/me/wojnowski/oidc4s/Algorithm.scala

```diff
@@ -0,0 +1,22 @@
+package me.wojnowski.oidc4s
+
+import cats.Order
+import cats.data.NonEmptySet
+import cats.implicits._
+
+sealed abstract class Algorithm(val name: String, val fullName: String) extends Product with Serializable
+
+// According to OIDC RFC, only RS256 should be supported
+object Algorithm {
+  case object Rs256 extends Algorithm(name = "RS256", fullName = "SHA256withRSA")
+  case object Rs384 extends Algorithm(name = "RS384", fullName = "SHA384withRSA")
+  case object Rs512 extends Algorithm(name = "RS512", fullName = "SHA512withRSA")
+
+  case class Other(override val name: String) extends Algorithm(name, fullName = name)
+
+  implicit val order: Order[Algorithm] = Order.by(_.name)
+
+  val supportedAlgorithms: NonEmptySet[Algorithm] = NonEmptySet.of(Rs256, Rs384, Rs512)
+
+  def fromString(s: String): Algorithm = supportedAlgorithms.find(_.name === s).getOrElse(Other(s))
```

Comment:
Most of it taken care of in https://github.com/jwojnowski/oidc4s/pull/57.

As for the link to the list of Java signature algorithms, I don't think it will add much — the library only supports a specific set of them.

However, I dove a bit deeper into the OIDC RFC and learned it is possible to use HMAC-based algorithms as well (and ECDSA I knew about earlier, but left that out for the time being). I guess it would be great to add those at some point (issue: https://github.com/jwojnowski/oidc4s/issues/58).